File: src/main/java/io/github/toberocat/guiengine/item/GuiItemManager.java
package io.github.toberocat.guiengine.item;

import io.github.toberocat.toberocore.item.ItemCore;
import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.HashMap;
import java.util.Map;

/**
 * Created: 21.07.2023
 *
 * @author Tobias Madlberger (Tobias)
 */
public class GuiItemManager {
    private final @NotNull Map<String, ItemStack> knownItems = new HashMap<>();

    public GuiItemManager(@NotNull FileConfiguration config) {
        ConfigurationSection section = config.getConfigurationSection("items");
        if (section == null)
            return;
        for (String key : section.getKeys(false)) {
            ConfigurationSection itemSection = section.getConfigurationSection(key);
            if (itemSection == null)
                continue;

            knownItems.put(key, ItemCore.create(itemSection));
        }
    }

    public @Nullable ItemStack getItem(@NotNull String name) {
        return knownItems.get(name);
    }

    public @NotNull Map<String, ItemStack> getKnownItems() {
        return knownItems;
    }
}

File: src/main/kotlin/io/github/toberocat/guiengine/item/GuiItemManager.kt
package io.github.toberocat.guiengine.item

import io.github.toberocat.toberocore.item.ItemCore
import org.bukkit.configuration.file.FileConfiguration
import org.bukkit.inventory.ItemStack

/**
 * GuiItemManager is responsible for managing and providing access to known GUI items.
 *
 *
 * Created: 21.07.2023
 * Author: Tobias Madlberger (Tobias)
 */
class GuiItemManager(config: FileConfiguration) {
    /**
     * A map that associates names of GUI items with their corresponding ItemStack representations.
     */
    private val knownItems: MutableMap<String, ItemStack> = HashMap()

    init {
        val section = config.getConfigurationSection("items")
        for (key in section?.getKeys(false) ?: emptySet()) {
            val itemSection = section?.getConfigurationSection(key) ?: continue
            knownItems[key] = ItemCore.create(itemSection)
        }
    }

    /**
     * Retrieves the ItemStack representation of the GUI item with the given name.
     *
     * @param name The name of the GUI item to retrieve.
     * @return The ItemStack representation of the GUI item if found, or null if not present in the manager.
     */
    operator fun get(name: String): ItemStack? = knownItems[name]

    /**
     * Returns a map containing all the known GUI items, where keys are the names of the items,
     * and values are their corresponding ItemStack representations.
     *
     * @return A map of known GUI items with names as keys and ItemStacks as values.
     */
    fun getKnownItems(): Map<String, ItemStack> = knownItems
}
